module_5/oop_classes.py

```diff
@@ -0,0 +1,171 @@
+        """
+        Method to generate random integer
+        :return: generated integer from 0 to 10
+        """
+        return random.randint(0, 10)
+
+
+    def get_user_number(self):
+        """
+        Method to get user's guessed number
+        :return: user's guessed number
+        """
+        print("\nYour choice will be published in next edition")
+        guess = input("""\nPlease enter your number from 0 to 10: """)
+        return guess
+
+    def success_of_fail(self,  guessed, randomed):
+        """
+        Method to compare if generated number matches with guessed
+        :param guessed: user's guessed number
+        :param randomed: generated number
+        :return: message if guessed or not
+        """
+        if randomed == int(guessed):
+            msg = "Yehoo! You're lucky today"
+        else:
+            msg = "Ohh, try another time"
+        return msg
+
+
+class PdqeNewsPaperSite(CommonMethods):
+    """
+    Class that gets information from user and puts it into txt file
+    """
+
+    def __init__(self, output_file):
+        self.output_file_name = output_file
+
+    news = News()
+    ads = Ads()
+    luck = CheckYourLuck()
+    common_methods = CommonMethods()
+
+    print("*" * 20 + f" Welcome to PDQE Newspaper site " + "*" * 20)
+    print("""\nPlease choose category for your message:
+            1 - News
+            2 - Advertising
+            3 - Check your luck""")
+
+    def get_category(self):
+        """
+        Method to get category for user's input
+        :return: 
+        """
+        val = input("\nEnter your value: ")
+        return val
+
+    def user_input(self):
+        """
+        Method to populate file with user's data
+        """
+        val = self.get_category()
+        if int(val) == 1:
+            title = f"\n" + "*" * 20 + f" News " + "*" * 20
+            self.news_text, self.news_city = self.news.get_news()
+            self.common_methods.populate_output_file(title,
+                                                     f"News: {self.news_text}",
+                                                     f"City: {self.news_city}",
+                                                     f"Publish date: {self.common_methods.get_current_date()}",
+                                                     f"{self.output_file_name}")
+
+        elif int(val) == 2:
+            title = f"\n" + "*" * 20 + f" Private Ads " + "*" * 20
+            self.ads_text, self.expiration_date = self.ads.get_ads()
+            self.common_methods.populate_output_file(title,
+                                                     f"Ads: {self.ads_text}",
+                                                     f"Expiration date: {self.expiration_date}",
+                                                     f"Days left: {self.ads.get_days_left(str(self.expiration_date))}",
+                                                     f"{self.output_file_name}")
+
+        else:
```

Comment:
Not quite right.
You specify "3 - Check your luck". But if you enter any number 0 or 4, 5 and so on, we will still go to step 3.
Try to remake this algorithm.

Reply:
fixed